modules/highgui/src/window_framebuffer.cpp

```diff
@@ -0,0 +1,410 @@
+  }
+
+  void FramebufferWindow::setMouseCallback(MouseCallback onMouse, void* userdata ){
+    std::cout  << "FramebufferWindow::setMouseCallback(...)" << std::endl;
+  }
+
+  std::shared_ptr<UITrackbar> FramebufferWindow::createTrackbar(
+      const std::string& name,
+      int count,
+      TrackbarCallback onChange,
+      void* userdata
+  ){
+    return nullptr;
+  }
+
+  std::shared_ptr<UITrackbar> FramebufferWindow::findTrackbar(const std::string& name){
+    return nullptr;
+  }
+  
+  const std::string& FramebufferWindow::getID() const  { 
+    std::cout  << "getID())" << std::endl; return FB_ID;
+  }
+
+  bool FramebufferWindow::isActive() const {
+    std::cout  << "isActive()" << std::endl; 
+    return true;
+  }
+
+  void FramebufferWindow::destroy() {
+    std::cout  << "destroy()" << std::endl;
+  }
+
+  int FramebufferBackend::OpenInputEvent()
+  {
+    int fd;
+    fd = open("/dev/input/event1", O_RDONLY);
+    if (fd == -1) {
+        std::cerr << "ERROR_OPENING_INPUT\n";
+        return -1;
+    }
+    return fd;
+  }
+
+
+// !!##FramebufferBackend
+
+  int FramebufferBackend::fb_open_and_get_info()
+  {
+    int fb_fd = open("/dev/fb0", O_RDWR);
+    if (fb_fd == -1)
+    {
+      std::cerr << "ERROR_OPENING_FB\n";
+      return -1;
+    }
+
+    // Get fixed screen information
+    if (ioctl(fb_fd, FBIOGET_FSCREENINFO, &fix_info)) {
+      std::cerr << "ERROR_READING_FIX_INFO\n";
+      return -1;
+    }
+
+    // Get variable screen information
+    if (ioctl(fb_fd, FBIOGET_VSCREENINFO, &var_info)) {
+      std::cerr << "EERROR_READING_VAR_INFO\n";
+      return -1;
+    }
+
+    return fb_fd;
+  }
+
+  fb_var_screeninfo &FramebufferBackend::getVarInfo()
+  {
+      return var_info;
+  }
+  fb_fix_screeninfo &FramebufferBackend::getFixInfo()
+  {
+    return fix_info;
+  }
+  int FramebufferBackend::getFramebuffrerID()
+  {
+    return framebuffrer_id;
+  }
+  int FramebufferBackend::getFBwidth()
+  {
+    return fb_w;
+  }
+  int FramebufferBackend::getFBheight()
+  {
+    return fb_h;
+  }
+  int FramebufferBackend::getFBXOffset()
+  {
+    return x_offset;
+  }
+  int FramebufferBackend::getFBYOffset()
+  {
+    return y_offset;
+  }
+  int FramebufferBackend::getFBbpp()
+  {
+    return bpp;
+  }
+  int FramebufferBackend::getFBLineLength()
+  {
+    return line_length;
+  }
+  unsigned char* FramebufferBackend::getFBPointer()
+  {
+    return fbPointer;
+  }
+  Mat& FramebufferBackend::getBackgroundBuff()
+  {
+    return backgroundBuff;
+  }
+
+
+
+
+
+  FramebufferBackend::FramebufferBackend()
+  {
+    std::cout  << "FramebufferBackend()" << std::endl;
+    framebuffrer_id = fb_open_and_get_info();
+    std::cout  << "FramebufferWindow():: id " << framebuffrer_id << std::endl;
+    
+    if(framebuffrer_id == -1){
+      fb_w = 0;
+      fb_h = 0;
+      y_offset = 0;
+      x_offset = 0;
+      bpp = 0;
+      line_length = 0;
+
+      return;
+    }
+    
+    fb_w = var_info.xres;
```

Comment:
We need to allow only supported FB configurations (check bits_per_pixel/grayscale/red/green/blue/transp/colorspace/...?) and log error message + return gracefully for unsupported configurations.

Reply:
1. Added conditions for FB format window_framebuffer.cpp (244-259)
2. Added image reformatting functionality window_framebuffer.cpp (61-81)
convertToShow - standard function from precomp.hpp (152-177)
3. An example of using multiple windows ([here](https://github.com/kozinove/framebuffer/tree/main/sample_fb_multiwindows)), a color image and a grayscale image [here](https://github.com/kozinove/framebuffer/tree/main/sample_fb_color_gs)